src/main/java/reactor/rabbitmq/Sender.java

```diff
@@ -582,7 +598,10 @@ private void handleAckNack(long deliveryTag, boolean multiple, boolean ack) {
                 }
             });
             state.set(SubscriberState.ACTIVE);
-            subscriber.onSubscribe(subscription);
+            if (Operators.validate(this.subscription, subscription)) {
```

Comment:
I'd guard the entire method with this, so that the listener is never added twice in case of a subscription race

Reply:
Thanks, done as you suggested